File: app/src/main/java/ezike/tobenna/myweather/data/local/WeatherDatabase.java
package ezike.tobenna.myweather.data.local;

import androidx.room.Database;
import androidx.room.RoomDatabase;
import androidx.room.TypeConverters;
import ezike.tobenna.myweather.data.local.dao.WeatherDao;
import ezike.tobenna.myweather.data.local.entity.WeatherResponse;

/**
 * @author tobennaezike
 */

@Database(entities = {WeatherResponse.class}, version = 1, exportSchema = false)
@TypeConverters({WeatherTypeConverter.class})
public abstract class WeatherDatabase extends RoomDatabase {

    public abstract WeatherDao weatherDao();
}

File: app/src/main/java/ezike/tobenna/myweather/data/local/db/WeatherDatabase.kt
package ezike.tobenna.myweather.data.local.db

import androidx.room.Database
import androidx.room.RoomDatabase
import androidx.room.TypeConverters
import ezike.tobenna.myweather.data.model.WeatherResponse


@Database(entities = [WeatherResponse::class], version = 1, exportSchema = false)
@TypeConverters(WeatherTypeConverter::class)
abstract class WeatherDatabase : RoomDatabase() {
    abstract fun weatherDao(): WeatherDao?
}
